### PR TITLE
Author managerment, Book Detail, Create

### PR DESCRIPTION
**Pull request by lesang458**

### CODE REVIEW

File: assignment-project/src/app/core/http/author.service.ts [hunk 1]
```diff
@@ -7,16 +7,28 @@ import { environment } from 'src/environments/environment';
...
+    return this.http.delete<void>(`${this.baseUrl}${id}`);
+  }
+
+  updateAuthor(id: number, author: Author): Observable<void> {
+    return this.http.put<void>(`${this.baseUrl}${id}`, author);
+  }
+
+  addAuthor(author: Author): Observable<void> {
+    return this.http.post<void>(`${this.baseUrl}`, author);
+  }
```
Comment (tranmanhlinh) [inline assignment-project/src/app/core/http/author.service.ts:33]: Add access modifier.

File: assignment-project/src/app/core/http/book.service.ts [hunk 2]
```diff
@@ -50,4 +50,20 @@ export class BookService {
...
+    return this.http.post<Book>(`${this.baseUrl}`, book);
+  }
+
+  updateBook(id: number, book: Book): Observable<Book> {
+    return this.http.put<Book>(`${this.baseUrl}${id}`, book);
+  }
+
+  deleteBook(id: number): Observable<void> {
+    return this.http.delete<void>(`${this.baseUrl}${id}`);
+  }
```
Comment (tranmanhlinh) [inline assignment-project/src/app/core/http/book.service.ts:68]: Same here.

File: assignment-project/src/app/core/http/category.service.ts [hunk 3]
```diff
@@ -7,22 +7,22 @@ import { environment } from 'src/environments/environment';
...
         if (currentIndex === selectedCategories.length - 1) {
-          return accumulator + currentValue.id;
+          return `${accumulator}${currentValue.id}`;
         }
-        return accumulator + currentValue.id + ',';
-      }, '');
-      params = params.append('selectedCategories', categoriesIds);
+        return `${accumulator}${currentValue.id},`;
+      }, ``);
+      params = params.append('selectedCategories', categoryIds);
```
Comment (tranmanhlinh) [inline assignment-project/src/app/core/http/category.service.ts:22]: I see that you use these lines of code very often. Please consider a Helper for this.

File: assignment-project/src/app/core/interceptors/token.interceptor.ts [hunk 4]
```diff
@@ -14,7 +14,7 @@ export class TokenInterceptor implements HttpInterceptor {
   constructor() {}
 
   intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
-    const token = 'Bearer eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJuYW1laWQiOiIxIiwidW5pcXVlX25hbWUiOiJtYW5hZ2VyMSIsInJvbGUiOiJNQU5BR0VSIiwibmJmIjoxNTk1NDg4MDQxLCJleHAiOjE1OTU5MjAwNDEsImlhdCI6MTU5NTQ4ODA0MX0.DNJ5O2C5xfoF99H0rR68-9_4ckLazHuSZ-wk-CyuAeyaMI-7ichJA8ympO87jKCYSkV_9Y01QFmhgiGUi7Fx4Q';
+    const token = 'Bearer eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJuYW1laWQiOiIxIiwidW5pcXVlX25hbWUiOiJtYW5hZ2VyMSIsInJvbGUiOiJNQU5BR0VSIiwibmJmIjoxNTk1OTIwNjIzLCJleHAiOjE1OTYzNTI2MjMsImlhdCI6MTU5NTkyMDYyM30.39H1we_GXLp65ZTbEH9GL8YAEUmYItoDhvCgmnXKf_swOkXC-t3eQu-bPpigkd_Tr_UeUG3zUuiQLGHAZSteDA';
```
Comment (tranmanhlinh) [inline assignment-project/src/app/core/interceptors/token.interceptor.ts:17]: Store a bearer token in localstorage.

File: assignment-project/src/app/core/services/alertify.service.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+  message(message: string): void {
+    alertify.message(message);
+}
+
+  alert(message: string): void {
+    alertify.alert(message);
+  }
+
```
Comment (tranmanhlinh) [inline assignment-project/src/app/core/services/alertify.service.ts:37]: Use access modifier.

File: assignment-project/src/app/core/http/author.service.ts (hunk 1 above)
Comment (lesang458) [reply assignment-project/src/app/core/http/author.service.ts:33]: done

File: assignment-project/src/app/core/http/book.service.ts (hunk 2 above)
Comment (lesang458) [reply assignment-project/src/app/core/http/book.service.ts:69]: done

File: assignment-project/src/app/core/http/category.service.ts (hunk 3 above)
Comment (lesang458) [reply assignment-project/src/app/core/http/category.service.ts:18]: done

File: assignment-project/src/app/core/interceptors/token.interceptor.ts (hunk 4 above)
Comment (lesang458) [reply assignment-project/src/app/core/interceptors/token.interceptor.ts]: done

File: assignment-project/src/app/core/services/alertify.service.ts (hunk 5 above)
Comment (lesang458) [reply assignment-project/src/app/core/services/alertify.service.ts]: done

**Pull request merged**